packages/core/admin/admin/src/utils/getFetchClient.ts

```diff
@@ -111,20 +110,107 @@ type FetchClient = {
  * };
  * ```
  */
-const getFetchClient = (defaultOptions: AxiosRequestConfig = {}): FetchClient => {
-  instance.defaults.baseURL = window.strapi.backendURL;
-  return {
-    get: (url, config) =>
-      instance.get(normalizeUrl(url), {
-        ...defaultOptions,
-        ...config,
-      }),
-    put: (url, data, config) =>
-      instance.put(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    post: (url, data, config) =>
-      instance.post(normalizeUrl(url), data, { ...defaultOptions, ...config }),
-    del: (url, config) => instance.delete(normalizeUrl(url), { ...defaultOptions, ...config }),
+const getFetchClient = (defaultOptions: FetchConfig = {}): FetchClient => {
+  const baseURL = defaultOptions.baseURL;
+  const backendURL = window.strapi.backendURL;
+  const headers = new Headers({
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${getToken()}`,
+  });
+
+  // Add a response interceptor to return the response
+  const responseInterceptor = async <TData = unknown>(
+    response: Response
+  ): Promise<FetchResponse<TData>> => {
+    const result = await response.json();
+    if (result.error) {
+      throw new FetchError(result.error.message, result);
+    }
+    return {
+      data: result,
+    };
+  };
+
+  const paramsSerializer = <Param = unknown>(url: string, params?: Param) => {
+    if (params) {
+      const serializedParams = qs.stringify(params, { encode: false });
+      return `${url}?${serializedParams}`;
+    }
+    return url;
+  };
+
+  const addBaseUrl = (url: FetchURL) => {
+    if (baseURL) {
+      return `${backendURL}${baseURL}${url}`;
+    }
+    return `${backendURL}${url}`;
+  };
+
+  const fetchClient: FetchClient = {
+    get: async <TData, R = FetchResponse<TData>>(
+      url: string,
+      options?: FetchOptions
+    ): Promise<R> => {
+      const response = await fetch(
+        paramsSerializer(addBaseUrl(normalizeUrl(url)), options?.params),
+        {
+          signal: options?.signal ?? defaultOptions.signal,
+          method: 'GET',
+          headers,
+        }
+      );
+      return responseInterceptor<TData>(response) as Promise<R>;
```

Comment:
How come we have to cast? If we're casting, we should write a comment, other people will have the same questions

Reply:
we're gonna fix this.